packages/node/src/integrations/localvariables/localvariables-async.ts

```diff
@@ -0,0 +1,287 @@
+    } else if (prop?.value?.value || prop?.value?.description) {
+      variables = { ...variables, ...unrollOther(prop) };
+    }
+  }
+
+  return variables;
+}
+
+/**
+ * Adds local variables to exception frames
+ *
+ * Default: 50
+ */
+export class LocalVariablesAsync implements Integration {
+  public static id: string = 'LocalVariablesAsync';
+
+  public readonly name: string = LocalVariablesAsync.id;
+
+  private readonly _cachedFrames: LRUMap<string, FrameVariables[]> = new LRUMap(20);
+  private _rateLimiter: RateLimitIncrement | undefined;
+  private _shouldProcessEvent = false;
+
+  public constructor(private readonly _options: Options = {}) {}
+
+  /**
+   * @inheritDoc
+   */
+  public setupOnce(_addGlobalEventProcessor: (callback: EventProcessor) => void, _getCurrentHub: () => Hub): void {
+    // noop
+  }
+
+  /** @inheritdoc */
+  public setup(client: NodeClient): void {
+    const clientOptions = client.getOptions();
+
+    if (!clientOptions.includeLocalVariables) {
+      return;
+    }
+
+    import('node:inspector/promises')
+      .then(async inspector => {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const { Session } = inspector as any;
+        return this._startDebugger(Session, clientOptions);
+      })
+      .catch(e => {
+        logger.error('Failed to load inspector API', e);
+      });
+  }
+
+  /** @inheritdoc */
+  public processEvent(event: Event): Event {
+    if (this._shouldProcessEvent) {
+      return this._addLocalVariables(event);
+    }
+
+    return event;
+  }
+
+  /** Start and configures the debugger to capture local variables */
+  private async _startDebugger(Session: Session, options: NodeClientOptions): Promise<void> {
+    const session = new Session();
+    session.connect();
+
+    let isPaused = false;
+
+    session.on('Debugger.resumed', () => {
+      isPaused = false;
+    });
+
+    session.on('Debugger.paused', async event => {
+      isPaused = true;
+
+      this._handlePaused(session, options.stackParser, event.params as PausedExceptionEvent)
+        .then(async () => {
+          if (isPaused) {
+            // After the pause work is complete, resume execution or the exception context memory is leaked
+            await session.post('Debugger.resume');
+          }
+        })
+        .catch(_ => {
+          //
+        });
+    });
+
+    await session.post('Debugger.enable');
+
+    const captureAll = this._options.captureAllExceptions !== false;
+    await session.post('Debugger.setPauseOnExceptions', { state: captureAll ? 'all' : 'uncaught' });
+
+    if (captureAll) {
+      const max = this._options.maxExceptionsPerSecond || 50;
+
+      this._rateLimiter = createRateLimiter(
+        max,
+        async () => {
+          logger.log('Local variables rate-limit lifted.');
+          await session.post('Debugger.setPauseOnExceptions', { state: 'all' });
+        },
```

Comment:
This will create an unnecessary async scope. Every async scope in node.js has a performance cost.

```suggestion
        () => {
          logger.log('Local variables rate-limit lifted.');
          return session.post('Debugger.setPauseOnExceptions', { state: 'all' });
        },
```